sources/packages/backend/workflow/BPMN/camunda-8/assessment-gateway.bpmn

```diff
@@ -174,55 +174,55 @@
       <bpmn:outgoing>Flow_1lbai2h</bpmn:outgoing>
       <bpmn:outgoing>Flow_0bo0mn4</bpmn:outgoing>
     </bpmn:exclusiveGateway>
-    <bpmn:callActivity id="retrieve-supporting-info-for-parent-1-task" name="Retrieve supporting info for parent 1">
+    <bpmn:callActivity id="retrieve-supporting-info-for-parent-1-subprocess" name="Retrieve supporting info for parent 1">
       <bpmn:extensionElements>
         <zeebe:calledElement processId="supporting-user-information-request" propagateAllChildVariables="false" />
         <zeebe:ioMapping>
           <zeebe:input source="=createdSupportingUsersIds[1]" target="supportingUserId" />
-          <zeebe:input source="=append(parentSubprocesses, &#34;retrieveSupportingInfoParent1Subprocess&#34;)" target="parentSubprocesses" />
+          <zeebe:input source="=[&#34;retrieve-supporting-info-for-parent-1-subprocess&#34;]" target="parentSubprocesses" />
           <zeebe:output source="=supportingUserId" target="parent1SupportingUserId" />
           <zeebe:output source="=totalIncome" target="supportingUserParent1TotalIncome" />
-          <zeebe:output source="=&#34;retrieveSupportingInfoParent1Subprocess&#34;" target="retrieveSupportingInfoParent1Subprocess" />
+          <zeebe:output source="=&#34;retrieve-supporting-info-for-parent-1-subprocess&#34;" target="retrieve_supporting_info_for_parent_1_subprocess_passthrough" />
         </zeebe:ioMapping>
       </bpmn:extensionElements>
       <bpmn:incoming>Flow_1u5ion6</bpmn:incoming>
       <bpmn:outgoing>Flow_1g59sq3</bpmn:outgoing>
     </bpmn:callActivity>
-    <bpmn:callActivity id="retrieve-supporting-info-for-parent-2-task" name="Retrieve supporting info for parent 2">
+    <bpmn:callActivity id="retrieve-supporting-info-for-parent-2-subprocess" name="Retrieve supporting info for parent 2">
       <bpmn:extensionElements>
         <zeebe:calledElement processId="supporting-user-information-request" propagateAllChildVariables="false" />
         <zeebe:ioMapping>
           <zeebe:input source="=createdSupportingUsersIds[2]" target="supportingUserId" />
-          <zeebe:input source="=append(parentSubprocesses, &#34;retrieveSupportingInfoParent2Subprocess&#34;)" target="parentSubprocesses" />
+          <zeebe:input source="=[&#34;retrieve-supporting-info-for-parent-2-subprocess&#34;]" target="parentSubprocesses" />
```

Comment:
Does `parentSubprocesses` still needs to be an array?

Reply:
It does if we are still willing to support one more level of hierarchy.

Reply:
Got it 👍